src/main/app/windowManager.js

```diff
@@ -197,10 +226,73 @@ class WindowManager extends EventEmitter {
     return result
   }
 
+  /**
+   * Find the best window to open the files in.
+   *
+   * @param {string[]} fileList File full paths.
+   * @returns {{windowId: string, fileList: string[]}[]} An array of files mapped to a window id or null to open in a new window.
+   */
+  findBestWindowToOpenIn (fileList) {
+    if (!fileList || !fileList.length) return []
+    const { windows } = this
+    const lastActiveEditorId = this.getActiveEditorId() // editor id or null
+
+    if (this.windowCount <= 1) {
+      return [ { windowId: lastActiveEditorId, fileList } ]
+    }
+
+    // Array of scores, same order like fileList.
+    let best = null
+    for (const window of windows.values()) {
+      if (window.type === WindowType.EDITOR) {
+        const score = window.getCandidatesScore(fileList)
+        if (!best) {
+          best = score
+        } else {
+          for (let i = 0; i < best.length; ++i) {
+            // Update score only if the file is not already opened.
+            if (best[i].score !== -1 && best[i].score < score[i].score) {
+              best[i] = score[i]
+            }
+          }
+        }
+      }
+    }
+
+    const buf = []
+    for (let i = 0; i < best.length; ++i) {
+      let { id: windowId, score } = best[i]
+
+      if (score === -1) {
+        // Skip files that already opened.
+        continue
+      } else if (score === 0) {
+        // There is no best window to open the file(s) in.
+        windowId = lastActiveEditorId
+      }
+
+      let item = buf.find(w => w.windowId === windowId)
+      if (!item) {
+        item = { windowId, fileList: [] }
+        buf.push(item)
+      }
+      item.fileList.push(fileList[i])
```

Comment:
What's the relationship between fileList[i] and best[i]? why only push fileList[i]?

Reply:
Please see above. `best` stores the path score of the best matching window without this helper array we would change the `fileList` array/object.